homeassistant/components/device_tracker/config_entry.py

```diff
@@ -153,8 +154,7 @@ def handle_device_event(ev: Event) -> None:
         if (entity_id := ent_reg.async_get_entity_id(DOMAIN, *unique_id)) is None:
             return
 
-        if (entity_entry := ent_reg.async_get(entity_id)) is None:
-            return
+        entity_entry = ent_reg.entities[entity_id]
```

Comment:
The removed code had an impossible check since we know that the entity_id is registered in the entity registry. The new code cleans this case up.